[VLLM][TDESC] Unified attention compilation errors for bf16 and fp8

**Last passing commit:** c4ee23291342afa6f4956cdaf658ec1270917827 (https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/23458365788)
- unified attention fp8 still fails in this run due to a different error tracked by https://github.com/intel/intel-xpu-backend-for-triton/issues/6419

**First failing commit:** 7a978e6ebdb7b1c747c9bb0f2ec2fb693195119c (https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/23455560801)

**Error:**
```
/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/vllm/vllm/attention/ops/triton_unified_attention.py:57:1: error: Failures have been detected while processing an MLIR pass pipeline
def kernel_unified_attention_2d(
^
/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/vllm/vllm/attention/ops/triton_unified_attention.py:57:1: note: Pipeline failed while executing [`TritonIntelTensorDescToBlockPointer` on 'builtin.module' operation]: reproducer generated at `std::errs, please share the reproducer above with Triton project.`
Traceback (most recent call last):
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/benchmarks/vllm/unified_attention/unified_attention_benchmark.py", line 394, in <module>
    _benchmark_attention.run(show_plots=False, print_data=True)
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/pyenv/versions/3.12.7/lib/python3.12/site-packages/triton_kernels_benchmark/benchmark_testing.py", line 575, in run
    df = self._run(bench, args.reports, show_plots, print_data, mark_args=args, run_counter=run_counter,
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/pyenv/versions/3.12.7/lib/python3.12/site-packages/triton_kernels_benchmark/benchmark_testing.py", line 500, in _run
    ret = self.fn(**x_args, **{bench.line_arg: y}, **bench.args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/benchmarks/vllm/unified_attention/unified_attention_benchmark.py", line 349, in benchmark
    benchmark_suite.assert_close(triton_fn, torch_fn, atol=atol, rtol=rtol, err_msg='triton to torch')
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/pyenv/versions/3.12.7/lib/python3.12/site-packages/triton_kernels_benchmark/benchmark_testing.py", line 335, in assert_close
    triton_assert_close(x_fn(), y_fn(), atol, rtol, err_msg)
                        ^^^^^^
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/benchmarks/vllm/unified_attention/unified_attention_benchmark.py", line 326, in triton_fn
    unified_attention(
  File "/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/vllm/vllm/attention/ops/triton_unified_attention.py", line 928, in unified_attention
```

**The error only applies to the version of the unified attention kernel that is patched to use tensor descriptors**.
Maybe start by taking a look at the patch (`benchmarks/vllm/unified_attention/unified_attention.patch`)?

--- a/test/Triton/Intel/TensorDescToBlockPointer/loop.mlir
+++ b/test/Triton/Intel/TensorDescToBlockPointer/loop.mlir
@@ -281,4 +281,43 @@ module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 4 : i32, "ttg.thr
   // CHECK:          scf.yield [[ARG4]] : !tt.ptr<tensor<8x128xf32, #blocked>>
   // CHECK:        }
 
+  // COM: Regression: rewriting the loop-carried descriptor can retag a later
+  // COM: make_tensor_descriptor result to tensor pointer type via type
+  // COM: propagation.
+  tt.func public @retagged_make_tensor_desc_in_loop(%arg0: !tt.ptr<f16>, %arg1: i32, %arg2: i32) {
+    %c0_i32 = arith.constant 0 : i32
+    %c1_i32 = arith.constant 1 : i32
+    %c4_i32 = arith.constant 4 : i32
+    %c8_i32 = arith.constant 8 : i32
+    %c1_i64 = arith.constant 1 : i64
+    %arg2_i64 = arith.extsi %arg2 : i32 to i64
+    %init_desc = tt.make_tensor_descriptor %arg0, [%arg1, %arg2], [%arg2_i64, %c1_i64] : <f16>, <tensor<16x32xf16>>
+
+    %final_desc = scf.for %iv = %c0_i32 to %c4_i32 step %c1_i32 iter_args(%iter_desc = %init_desc) -> (!tt.tensordesc<tensor<16x32xf16>>) : i32 {
+      %inner_desc = tt.make_tensor_descriptor %arg0, [%arg1, %arg2], [%arg2_i64, %c1_i64] : <f16>, <tensor<16x32xf16>>
+      %loaded = tt.descriptor_load %iter_desc[%c8_i32, %c0_i32] : !tt.tensordesc<tensor<16x32xf16>> -> tensor<16x32xf16, #blocked>
+      %is_zero = arith.cmpi eq, %iv, %c0_i32 : i32
+      %next_desc = arith.select %is_zero, %iter_desc, %inner_desc : !tt.tensordesc<tensor<16x32xf16>>
+      scf.yield %next_desc : !tt.tensordesc<tensor<16x32xf16>>
+    }
+
+    %loaded_out = tt.descriptor_load %init_desc[%c8_i32, %c0_i32] : !tt.tensordesc<tensor<16x32xf16>> -> tensor<16x32xf16, #blocked>
+    tt.return
+  }
+  // CHECK:      tt.func public @retagged_make_tensor_desc_in_loop({{.*}}) {
+  // CHECK-NOT:    tt.make_tensor_descriptor
+  // CHECK-NOT:    tt.descriptor_load
+  // CHECK:        [[INIT_PTR:%.+]] = tt.make_tensor_ptr {{.*}} : <tensor<16x32xf16, #blocked>>
+  // CHECK:        [[FOR_RES:%.+]] = scf.for [[IV:%.+]] = {{.*}} iter_args([[ITER_DESC:%.+]] = [[INIT_PTR]]) -> (!tt.ptr<tensor<16x32xf16, #blocked>>) : i32 {
+  // CHECK:          [[INNER_PTR:%.+]] = tt.make_tensor_ptr {{.*}} : <tensor<16x32xf16, #blocked>>
+  // CHECK:          [[ADV_IN:%.+]] = tt.advance [[ITER_DESC]], {{.*}} : <tensor<16x32xf16, #blocked>>
+  // CHECK:          tt.load [[ADV_IN]] {boundaryCheck = array<i32: 0, 1>, padding = 1 : i32} : !tt.ptr<tensor<16x32xf16, #blocked>>
+  // CHECK:          [[NEXT_PTR:%.+]] = arith.select {{.*}}, [[ITER_DESC]], [[INNER_PTR]] : !tt.ptr<tensor<16x32xf16, #blocked>>
+  // CHECK:          scf.yield [[NEXT_PTR]] : !tt.ptr<tensor<16x32xf16, #blocked>>
+  // CHECK:        }
+  // CHECK:        [[ADV_OUT:%.+]] = tt.advance [[INIT_PTR]], {{.*}} : <tensor<16x32xf16, #blocked>>
+  // CHECK:        tt.load [[ADV_OUT]] {boundaryCheck = array<i32: 0, 1>, padding = 1 : i32} : !tt.ptr<tensor<16x32xf16, #blocked>>
+  // CHECK:        tt.return
+  // CHECK:      }
+
 }

